testnet/stacks-node/Cargo.toml

```diff
@@ -21,6 +21,7 @@ base64 = "0.12.0"
 backtrace = "0.3.50"
 libc = "0.2"
 slog = { version = "2.5.2", features = [ "max_level_trace" ] }
+ctrlc = { version = "3.1.7", features = [ "termination" ] }
```

Comment:
So, I took the time to look through this crate and its dependencies.  Here's what I found:

* `ctrlc` depends on `nix 0.18`, which in turn depends on `libc 0.2.72`.  We already have `libc 0.2.82` as a transitive dependency used consistently through blockstack-core and stacks-node's dependencies, so this crate definitely brings in a lot of extra code.  If it used `nix 0.20`, this problem would go away.
* `ctrlc`'s use of `nix` is questionable to me, because it's just using it to get access to signal handlers on UNIX platforms.  This is already provided by `libc`.  Using `nix` is odd here, because the value-add of having `nix` is to provide safe wrappers around `libc`'s unsafe code.  This benefit is lost on this crate, because this crate does a ton of necessarily-unsafe things with signal handling.  So, I'm not really a fan of this crate's packaging choices -- it should have just used `libc`.
* `ctrlc` commits a couple of somewhat bad library-writer sins that could come back to bite us later:
   * It has global state of its own -- namely, it doesn't let you register more than one signal handler via a global flag.  I'm willing to be lenient on this specific case, however, because signal handlers are already global state, and it's already bad programming practice for libraries to add their own signal handlers.
   * It uses the "signal pipe" trick to forward signals from a signal handler into the appropriate thread to handle them (a common tactic for dealing with signals in multi-threaded Unix programs), BUT, instead of allowing the application to handle incoming signal data from this pipe, it instead starts a thread internally to handle it.  This thread, in turn, calls the user-given callback function.
   * The fact that it uses a pipe _at all_ is odd to me, because Rust has safer IPC constructions for this purpose (like channels).  The code that forwards the signal data through the pipe is written without any error checking (!!), but it does at least have a lot of `TODO` comments to remind the author to fix them (for whatever that's worth).

Unfortunately, I don't think there's anything better for signal handling right now, short of writing our own..  The one redeeming feature of `ctrlc` is that it handles the Windows equivalent of signals, whereas none of the alternatives I've looked at do so:

* I looked at `signal_hook`, which doesn't depend on anything we don't already depend on, but it only works on Unix.
* I looked at `chan_signal`, but it's obsolete and superseded by `crossbeam-channel`, which has a crap-ton of dependencies.
* I considered just writing a small module for signal handling, but this again doesn't address the lack of Windows support (which is sadly much more complicated than Unix support).

Reply:
Thanks for auditing this @jcnelson!